src/db.rs

```diff
@@ -266,6 +267,51 @@ impl DbInner {
         }
     }
 
+    #[allow(clippy::panic)]
+    pub async fn write_batch_with_options(&self, batch: &WriteBatch, options: &WriteOptions) {
+        self.maybe_apply_backpressure().await;
```

Comment:
It's probably best to either fail, or just return immediately on an empty batch. If the current wal/memtable is empty, and no other writes arrive the current implementation would just hang because the current wal/memtable would never be flushed (since nothing was written to it).

Reply:
Good call. Fixed here: 13d4ec50205c4a19bff0ddba4602ed1d0e848d08